File: 1/backend/gateway/src/controller/AuthController.js
import authService from '../service/AuthService.js'

class AuthController {
  async login(ctx) {
    const { username, password, environment } = ctx.validatedBody
    const traceLogger = ctx.traceLogger

    const result = await authService.login(username, password, environment, traceLogger)

    ctx.status = result.code
    ctx.body = {
      code: result.code,
      message: result.message,
      data: result.data || null,
      traceId: traceLogger.traceId,
      timestamp: Date.now()
    }
  }

  async logout(ctx) {
    const traceLogger = ctx.traceLogger
    const userId = ctx.state.user.id

    await authService.logout(userId, traceLogger)

    ctx.body = {
      code: 200,
      message: '登出成功',
      data: null,
      traceId: traceLogger.traceId,
      timestamp: Date.now()
    }
  }

  async currentUser(ctx) {
    const traceLogger = ctx.traceLogger
    const userId = ctx.state.user.id

    const result = await authService.getCurrentUser(userId, traceLogger)

    ctx.status = result.code || 200
    ctx.body = {
      code: result.code || 200,
      message: result.message || 'success',
      data: result.data || null,
      traceId: traceLogger.traceId,
      timestamp: Date.now()
    }
  }

  async changePassword(ctx) {
    const traceLogger = ctx.traceLogger
    const userId = ctx.state.user.id
    const { oldPassword, newPassword } = ctx.request.body

    const result = await authService.changePassword(userId, oldPassword, newPassword, traceLogger)

    ctx.status = result.code || 200
    ctx.body = {
      code: result.code || 200,
      message: result.message,
      data: null,
      traceId: traceLogger.traceId,
      timestamp: Date.now()
    }
  }
}

export default new AuthController()

File: 3/backend/src/main/java/com/research/sample/auth/controller/AuthController.java
package com.research.sample.auth.controller;

import com.research.sample.auth.context.TenantContext;
import com.research.sample.auth.entity.Role;
import com.research.sample.auth.entity.User;
import com.research.sample.auth.repository.UserRepository;
import com.research.sample.auth.service.AuthService;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.*;

import java.util.List;
import java.util.Map;

@RestController
@RequestMapping("/api/auth")
public class AuthController {

    private final AuthService authService;
    private final UserRepository userRepository;

    public AuthController(AuthService authService, UserRepository userRepository) {
        this.authService = authService;
        this.userRepository = userRepository;
    }

    @PostMapping("/login")
    public ResponseEntity<Map<String, Object>> login(@RequestBody Map<String, String> request) {
        Map<String, Object> result = authService.login(
                request.get("username"),
                request.get("password")
        );
        return ResponseEntity.ok(result);
    }

    @PostMapping("/register")
    public ResponseEntity<Map<String, Object>> register(@RequestBody Map<String, String> request) {
        Map<String, Object> result = authService.register(
                request.get("username"),
                request.get("password"),
                request.get("realName"),
                request.get("tenantId") != null ? Long.parseLong(request.get("tenantId")) : null,
                request.get("department")
        );
        return ResponseEntity.ok(result);
    }

    @GetMapping("/info")
    public ResponseEntity<Map<String, Object>> info() {
        Long userId = TenantContext.getUserId();
        if (userId == null) {
            return ResponseEntity.status(401).body(Map.of("message", "未登录"));
        }

        User user = userRepository.findById(userId)
                .orElseThrow(() -> new RuntimeException("用户不存在"));

        List<Role> roles = authService.getUserRoles(userId);

        return ResponseEntity.ok(Map.of(
                "userId", user.getId(),
                "username", user.getUsername(),
                "realName", user.getRealName(),
                "email", user.getEmail() != null ? user.getEmail() : "",
                "phone", user.getPhone() != null ? user.getPhone() : "",
                "tenantId", user.getTenantId() != null ? user.getTenantId() : 0,
                "department", user.getDepartment() != null ? user.getDepartment() : "",
                "roles", roles.stream().map(Role::getRoleCode).toList()
        ));
    }
}
